backend/apps/readings/analysis.py

```diff
@@ -6,6 +6,49 @@
 from .models import StudentResponse
 
 
+def get_responses_for_question(all_responses, question):
+    """
+    For a certain question, returns the set of responses as a dictionary with keys being the
+    context and values being nested dictionaries containing each response and their frequency.
+    :param all_responses: QuerySet of all student responses
+    :param question: string, question
+    :return: dictionary mapping strings to integers
+    """
+    responses = {}
+    for response in all_responses:
+        student_question = response.question.text
+        student_answer = response.response.lower()
+        question_context = response.context.text
+        if student_question == question:
+            if question_context not in responses:
```

Comment:
Take a look at the Counter class:
https://docs.python.org/3.7/library/collections.html#collections.Counter